packages/grid/x-data-grid/src/tests/rows.DataGrid.test.tsx

```diff
@@ -145,6 +146,77 @@ describe('<DataGrid /> - Rows', () => {
     });
   });
 
+  describe('columnType: actions with generic arguments', () => {
+    interface TestRowModel {
+      id: number;
+      foo: {
+        a: number;
+        b: number;
+      };
+      bar: {
+        a: string;
+        b: string;
+      };
+    }
+
+    const rows: TestRowModel[] = [{ id: 1, foo: { a: 1, b: 2 }, bar: { a: 'bar-a', b: 'bar-b' } }];
+
+    const TestCase = ({ ...other }: Partial<DataGridProps>) => {
+      const [id, setId] = React.useState<number | string>(-1);
+      const [value, setValue] = React.useState('');
+      const onClick = (newId: string | number, newValue: string) => {
+        setId(newId);
+        setValue(newValue);
+      };
+
+      const actionsColumn = React.useMemo((): GridActionsColDef<TestRowModel> => {
+        return {
+          field: 'actions',
+          type: 'actions',
+          getActions: ({ id: rowId, row }) => {
+            return [
+              <GridActionsCellItem
+                label="more"
+                onClick={() => onClick(rowId, row.bar.a)}
+                icon={<span />}
+              />,
+            ];
+          },
+        };
+      }, []);
+
+      return (
+        <div>
+          <div data-testid="id">{id || 'none'}</div>
```

Comment:
We usually don't use DOM elements to test if a function received a specific argument. Instead, we use spy functions.

Example:

https://github.com/mui/mui-x/blob/23a57c3665ff00675c976cc89f8aaeee54cefe69/packages/grid/x-data-grid-pro/src/tests/cellEditing.DataGridPro.new.test.tsx#L83-L89

Could you use this same approach here? It reduces the boilerplate.

Reply:
I think I can go ahead and pull the tests out of this file since the behavior is already tested and the compiler tests are covered elsewhere.